### ISSUE TITLE
Design: Create Style Guide Presentation (Radio Buttons/Checkboxes)

### ISSUE THREAD

**Issue by NilakshiS** [hackforla/tdm-calculator#1839]
### Dependency
- [x] #1850 ( Waiting on Stakeholder decision of which column dropdown will be chosen)

### Overview
The style guide currently does not mention any styling for radio buttons and check boxes. Add these to the Figma Design system and style guide presentation.

### Details
The style guide does not explicitly address the styling for radio buttons, though the Example Modal Dialog on the ["Dialog Boxes and Modals" page](https://docs.google.com/presentation/d/1I4q35NL2WW2RpksIyawhHCd8qJnrjme1ZDNIBxu24hQ/edit#slide=id.g1443c52839c_0_57) does implicitly show an example from the app.

However the example uses the "Navy Blue - Primary (`#002E6D`)" color for the radio buttons while the website uses the default browser styling for them. The default styling is used in several other UI elements such as checkboxes and dropdowns. 

So we need to decide if we should change the styling of these elements to keep it consistent with the colors of the design system or keep it browser default since it's the style users would already be familiar with.

### Action Items
- [x] Determine the styling for radio button, checkboxes and other similar UI elements:
  - [x] Research examples where it has been changed or kept the same.
  - [x] Decide on a style and create prototypes.
- [x] Add a section for these styles in the design system
- [x] Review with Product/ Dev/ Research and iterate if needed based on feedback.
- [x] If the style is different than dev, open issues for dev to update the design.
- [x] Update the style guide to add the new design.
- [ ] Once finalized, get Stakeholder sign-off via the stakeholder meeting slide deck.

### Resources
- [Google Slides Style Guide Presentation](https://docs.google.com/presentation/d/1I4q35NL2WW2RpksIyawhHCd8qJnrjme1ZDNIBxu24hQ/edit#slide=id.g2a6d1bfeb46_0_103)
- [Link to style guide in Figma](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=0-1&node-type=canvas&t=OFGKeYKJdGtVlx4i-0)
- [Presentation format](https://github.com/hackforla/tdm-calculator/wiki/Style-Guide#:~:text=calculator.wiki.git-,Presentation%20Format,-Add%20a%20custom)

**Comment by NilakshiS** [comment]
Iteration 2:
- Changed the checkboxes and radio buttons back to dev styles
- Changed the dropdown color to dev style but kept the increased padding

[Link to iteration 2 Figma](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=14846-2244&t=6h9ueSoHxKaFZ9V9-1)

<details><summary>Image for Iteration 2</summary>

![Image](https://github.com/user-attachments/assets/775aa52b-fe04-445b-a01f-4f887e25f0bc)

</details> 

Waiting on stakeholder feedback

**Comment by NilakshiS** [comment]
Add the default and Hover buttons to the slide explaining difference between them or add an asterisk on the "Hover" on the previous slide to indicate there is more to expalin on next slide.

**Comment by NilakshiS** [comment]
Current slide is approved by stakeholders, but we might need to add a multi-select dropdown from the filters to this list based on whatever's finalized

**Comment by NilakshiS** [comment]
Added the nested checkboxes from My Projects Page to the style guide presentation and design system on Figma:
![Image](https://github.com/user-attachments/assets/f718c364-e251-4f19-bcc4-fdf39e87bf66)

- [Link to slide on the style guide presentation](https://docs.google.com/presentation/d/1I4q35NL2WW2RpksIyawhHCd8qJnrjme1ZDNIBxu24hQ/edit#slide=id.g305f865959f_0_11)
- [Link to slide on design system in Figma](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=16061-4932&t=XEKMtITaNbHlCJ2M-1)

**Comment by thetanmancan** [comment]
Hi @NilakshiS. The nested checkboxes in the Dev site do not have horizontal scrolling. Should they? Or should we change the style guide to reflect the fact that the text within each option wraps to a second line when its length is longer than than width of the container?

**Comment by ExperimentsInHonesty** [comment]
@thetanmancan This issue has a 'waiting for stakeholder label' but the last comment you made is for NilakshiS.  I see this issue is in the deck with an image with a horizontal scroll... so I am confused.

**Comment by NilakshiS** [comment]
Hi @thetanmancan I referenced an old image for the design, but I see a horizontal scrollbar on the dev site. It's a bit inconsistent, there's one on the admin notes filter, however I don't see any on address filters. I can remove the horizontal scrollbar, but I'm not sure how we're planning on handling longer text, or do we not expect data longer than two lines? 

Here's some examples from the dev site where the longer text wraps:
<details><summary>Horizontal scrollbar on the admin notes</summary>
<p>
The checkbox size is the same here, but the wrapped text is middle aligned to the checkbox

![image](https://github.com/user-attachments/assets/38b45ffd-bf66-4166-91fc-477437a2f4a4)
</p>

<p>
The checkbox is smaller here,

![image](https://github.com/user-attachments/assets/3d8e2670-e673-4bf3-a6bf-ac6bdee37aa1)
</p>
</details> 

<details><summary>No scrollbar on address column</summary>

<p>
The checkbox size is the smaller here and the wrapped text is middle aligned to the checkbox

![image](https://github.com/user-attachments/assets/fd949524-d0a5-4532-b6ea-df1c60638a16)
</p>
</details>

**Comment by thetanmancan** [comment]
@NilakshiS I tested Admin Notes filter and did not see a horizontal scroll bar, even when I zoomed in on my browser.
<details><summary>Check it out</summary>

<img src=”https://github.com/user-attachments/assets/5adfb42d-123e-4339-9f96-78d72bd2b310” width=500px>
</details>

However, how the overlay behaves right now might not be a relevant conversation because I think how we're dealing with text strings in filter overlays is going to be changed. At least for some columns, there will be a 25-character limit in the width of the filter overlays according to #1651 , after which an ellipsis will be added and the rest of the text will be cut off. I will ask John at today's meeting about whether that limit is meant to apply to all column filters. If it is, the horizontal scroll bar won't be relevant, but even if it isn't, I think the text will still end up wrapping like it is now, in which case the horizontal scroll bar still won't be relevant, but I will have to confirm with John.

If we end up not having a horizontal scroll bar, we can remove it from the design system and I'll run that by the stakeholders in our next meeting. As for the vertical scroll bar, that's determined by the browser anyways, right? Or can our site dictate how that looks?

**Comment by thetanmancan** [comment]
Hi @entrotech. At one of our meetings, Bonnie requested in #1651 that the overlay for the Admin Notes filter only display 25 characters for item names, followed by an ellipsis. Will that same logic be applied to all filters?

**Comment by thetanmancan** [comment]
Hi @NilakshiS. Due to the advent of #2043, it seems like horizontal scroll bars will not be necessary for lists of nested checkboxes, so I removed the horizontal scrollbar from our design system mockup.
<details><summary>No horizontal scroll bar</summary>

![Image](https://github.com/user-attachments/assets/1780a430-0560-4f96-966d-6147adc9d465)
</details>

If you think the 25-character limit within lists of nested checkboxes is something we should include in the style guide, please let me know and I'll make an issue for it. Thanks.

**Comment by thetanmancan** [comment]
Hi @NilakshiS. I noticed while testing these styles in the Dev site that we use two different colors for each of the components listed: one upon hover and one upon selection. This highlights a number of problems with our style guide.

1. With the exception of dropdowns, none of the components in the "Radio buttons, checkboxes, and dropdowns" section of the style guide list a color for hover and a color for selected.
2. The color "Radio button default" that we have in Figma is not listed in the color palette.
3. The selection color for dropdowns is different than the one used for radio buttons and checkboxes. Is this inconsistency intentional?

Could you please reconcile these issues in the design system? I'll move this issue back to In Progress in the meanwhile. Thank you.

**Comment by NilakshiS** [comment]
Hi @thetanmancan thank you for the updates and making those changes! 
- I suppose the dropshadow got carried over from older designs and I didn't notice, thanks for catching that! 
- About the 25-character limit, I know this number was discussed in the meetings but I do not see it anywhere on the dev issue #2043 . If we're sure that this will be the character limit for all the nested checkboxes displays, we can add it to the style guide. I don't think it needs a new issue, it's related to this issue so I'll add it once the limit is confirmed.
- About the different colors and hover styles for these components, these are the default dev styles and we decided they looked good as they are, instead of using the colors from the color palette. Check the comment [here](https://github.com/hackforla/tdm-calculator/issues/1839#issuecomment-2387199617) for more info. The reason why I added "Radio button default" was to match these styles, these designs are not custom and left to dev implementation, so I decided to not add this to our color palette.
- However, now that I rechecked these, I realized that there has been a mistake, I was under the assumption that these are ReactJS styles (which would look the same across browsers) but instead these are browser native styles. The design as it currently looks is how it looks on Chrome Browser running on windows OS.
- The next steps could be either to customize these like how we did for dropdowns (that's the reason why they have a different selection color, check #1785) or remove these from the design system, since the design is dependent on the browser.
   - On side note, you are right about missing hover state designs, they are in mockups but I missed adding them to the slide. If we decide to customize, I'll add the hover state design for all of them.

**Comment by thetanmancan** [comment]
Hi @NilakshiS.
- In the comment section of #2043, I addressed John with a question about the specifics of the 25-character limit, which is detailed in the first comment of the issue. Once he answers, I'll let you know so that you can update the style guide.
- It seems pretty clear to me that we need to have a design for radio buttons and checkboxes across browsers, not only because it looks like radio buttons and checkboxes are grayed out when they are selected on Edge, but also because there should be some color consistency between dropdowns and radio buttons and checkboxes. Do you think this is something we need to run by production and development before developing mockups?
- I don't see any mockups for hover state designs in the Figma style guide, or at least not for radio buttons an checkboxes. Can you point them out to me?

**Comment by thetanmancan** [comment]
To address the above comment:
- Issues #2043 and #2063 will address the question of filter width. While filter width needs to be documented in our style guide, it is not within the scope of this issue.
- Issue #2072 has been written to account for the lack of hover state styles and the color inconsistencies between components. I've made this issue dependent on that issue and will be moving this issue to the ice box until that issue is resolved.

**Comment by ExperimentsInHonesty** [comment]
@NilakshiS Now that we have clarified scope, please provide update
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures or links* (if necessary): "Add any pictures or links that will help illustrate what you are working on."

* remember to add links to the top of the issue if they are going to be needed again.

**Comment by NilakshiS** [comment]
1. Progress:
- I created some mockups with different colors for radio buttons and checkboxes. We do not have a lot of choices for colors in our palette apart from greys and blues, both of which do not work well with these elements because it makes them look disabled or blends them with other UI elements on page. There's also green but it looks similar to the primary action buttons then.
<details><summary>Mockups with grey, blue, and green</summary>

![Image](https://github.com/user-attachments/assets/10aac043-fe73-4d96-bdc2-aa3fc0e4402b)

</details> 
 
- The only other choice would be to use the bright blue that we already use for dropdowns `#1967D2`, or stick to the blue that Chrome Browser uses `#0075FF`.

<details><summary>Dropdown blue vs. Chrome Blue for radio buttons and checkboxes</summary>

![Image](https://github.com/user-attachments/assets/6ed451f8-cb66-448d-bca2-af980bbf6ee6)

</details> 

- I recommend using the blue we already use for dropdowns.
- Created mockups for Hover states for checkboxes and radio buttons
<details><summary>Default and Hover states, the Hover states are on the right</summary>

![Image](https://github.com/user-attachments/assets/83aa6f42-c70b-4a94-9f9d-30126d0abd00)

</details> 

2. Blockers: No blockers, need feedback for the final color to choose
3. Availability: about 6 hours this week
4. ETA: 1 week
5. Pictures or links* (if necessary): [Link to these mockups on Figma](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=18111-6969&t=3SLCZoBPVVtG4L8N-1)

**Comment by ExperimentsInHonesty** [comment]
@Tristan Camacho and @NilakshiS re https://github.com/hackforla/tdm-calculator/issues/1839#issuecomment-2634753758

It looks like @NilakshiS is recommending the default blue for drop downs become the permanent check box and drop down color.  I cant remember if we decided we were going to run this through research to have them get a preference test.   Do you remember?

**Comment by thetanmancan** [comment]
@ExperimentsInHonesty I don’t recall bringing up the need to run our designs by research, though we can if you think that’s prudent.

**Comment by ExperimentsInHonesty** [comment]
Product needs to make a dev issue to create a class for the check box and drop down selection color. 
 `implement: make the default medium blue`, therby making it the same for everyone, by specifying it instead of the current situation which is windows gets medium blue, mac gets gray.  

@thetanmancan if design prefers the darker blue (I kind of like it but don't have a strong preference either way).  Then the question is just is the darker blue more pleasing or does it make it easier or harder for people to use the selectors.  And that can get answered by research doing a simple preference test through https://lysna.com/ and if we do go with the darker, then they would have to test it with users during normal usability testing once its in place.  Does design have a preference for the darker and want us to do this?

**Comment by marlenamellody** [comment]
**25-02-05 Design meeting feedback:** 

The team prefers the bright blue used for dropdowns `#1967D2`. This is because the grey option appears disabled, the dark blue in our design system blends with the column header, and the green closely resembles the primary action button. Currently, the suggested blue and the hover state blue are not part of our design system.

**Comment by ExperimentsInHonesty** [comment]
This is approved.  Dev will make an issue
